holidays/countries/thailand.py

```diff
@@ -11,157 +11,660 @@
+        # Sources:
+        #   https://th.wikipedia.org/wiki/วันปิยมหาราช
+        # วันปิยมหาราช
+        # Status: In-Use
+        # Started in 1911 (B.E. 2454)
+        rama_five_memorial = "Chulalongkorn Memorial Day"
+
+        if year >= 1911:
+            add_holiday(date(year, OCT, 23), rama_five_memorial)
+
+        # !!! HM King Bhumibol Adulyadej's Birthday Anniversary !!!
+        # Sources:
+        #   (Ministry of Culture 's wbm) http://tiny.cc/wa_mincul_nat_day
+        #   https://hilight.kapook.com/view/148862
+        # วันเฉลิมพระชนมพรรษา รัชกาลที่ 9 (1960-2016),
+        # วันคล้ายยวันเฉลิมพระชนมพรรษา รัชกาลที่ 9 (2017-Present)
+        # Status: In-Use
+        # Replaced Nataion Day (26 June) in 1960 (B.E. 2503) by Sarit Thanarat
+        # Confirmed as still in-use in 2017
+        rama_ix_birthday = "HM King Bhumibol Adulyadej's Birthday"
+
+        if year >= 1960:
+            add_holiday(date(year, DEC, 5), rama_ix_birthday)
+
+        # !!! Father's Day !!!
+        # Sources:
+        #   https://www.brh.go.th/index.php/2019-02-27-04-12-21/594-5-5
+        # วันพ่อแห่งชาติ
+        # Status: In-Use
+        # Starts in 1980 (B.E 2523)
+        # Technically, a replication of HM King Bhumibol Adulyadej's Birthday
+        #   but it's in the official calendar, so may as well have this here
+        thai_fathers_day = "Father's Day"
+
+        if year >= 1980:
+            add_holiday(date(year, DEC, 5), thai_fathers_day)
+
+        # !!! Constitution Day !!!
+        # Sources:
+        #   https://hilight.kapook.com/view/18208
+        #   (Bank of Thailand 's wbm) http://tiny.cc/wa_bot_1992
+        #   https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2475.aspx
+        # วันรัฐธรรมนูญ
+        # Status: In-Use
+        # Presumed to starts in 1932 (B.E. 2475) ???
+        # Last known official record is Bank of Thailand's in 1992 (B.E. 2535)
+        constitution_day = "Constitution Day"
+
+        if year >= 1932:
+            add_holiday(date(year, DEC, 10), constitution_day)
+
+        # !!! New Year's Eve !!!
+        # Sources:
+        #   (Bank of Thailand 's wbm) http://tiny.cc/wa_bot_1992
+        # วันสิ้นปี
+        # Status: In-Use
+        # Presumed to start in the present form in 1941 (B.E. 2484) ???
+        # Last known official record is Bank of Thailand's in 1992 (B.E. 2535)
+        # This has its own in-lieu trigger
+        new_years_eve = "New Year's Eve"
+
+        if year >= 1941:
+            self[date(year, DEC, 31)] = new_years_eve
+
+        ################################
+        #
+        # THAI LUNAR CALENDAR HOLIDAYS
+        #
+        ################################
+
+        """
+        See _ThaiLuniSolar in holidays/utils.py for more details
+        """
+        # !!! Makha Bucha !!!
+        # Sources:
+        #   https://www.onab.go.th/th/content/category/detail/id/73/iid/3403
+        # วันมาฆบูชา
+        # Status: In-Use
+        makha_bucha = "Makha Bucha"
+
```

Comment:
Need to check for START_YEAR <= year <= END_YEAR here or in corresponding `..._date()` functions

Reply:
Let's do both for safety's sake.